### ISSUE TITLE
Date and Datetime fields do not work for anonymous

### ISSUE THREAD

**Issue by pbauer** [collective/collective.easyform#203]
The pickadate-pattern is only part of the of the bundles `plone-editor-tools` and `plone-tinymce` which are only avaibale for logged-in users. 
Is there a way to work around this?

**Comment by tkimnguyen** [comment]
Copy them into the add-on?

**Comment by pbauer** [comment]
Copy what into the addon? The bundle, the pattern or only the registration?

**Comment by thet** [comment]
May we should peel out the pickadate pattern from plone-editor-tools and make it available as a seperate bundle? That's to be done in plone.staticresources.
Then collective.easyform could just enable that bundle via the registry.

**Comment by agitator** [comment]
I'm open for further splitting of bundles @pbauer remind me at the sprint ;-)
But please no generic auto-enable from easyform ...

**Comment by tkimnguyen** [comment]
> Copy what into the addon? The bundle, the pattern or only the registration?

I meant the pattern, but @thet ‘s idea is much better

**Comment by tkimnguyen** [comment]
Actually, I’d the pattern isn’t heavy, why not move it into the non logged in bundle?

**Comment by agitator** [comment]
one can always extend the condition for the bundle to something like `python: member is not None or context.portal_type in ('EasyForm')`

Depending on the usecase putting "everything" into a bundle doesn't make sense because it adds unnecessary load and therefore slows down the experience. We did that wrong with Plone for too long!

**Comment by tkimnguyen** [comment]
Is it ok to mention a non core content type? That dredges up the question: why not include EasyForm in Plone buildout.cfg but commented out, as used to be the case with PloneFormGen (maybe in the unified installer)

**Comment by davide-targa** [comment]
Maybe I misunderstood the issue (is it a datepicker problem?) but I just tried on a clean Plone 5.2 buildout with Python 3.7.3: Date and Datetime fields work as expected.

**Comment by thet** [comment]
@davide-targa probably only if you're logged-in, as the pickadate pattern is part of the plone-editor-tools bundle, which itself is only available to logged in users?

https://github.com/plone/plone.staticresources/blob/a3149d90ffe7ff8c0875bbb278243ae9cf02ae6a/src/plone/staticresources/static/plone-logged-in.js#L21

https://github.com/plone/plone.staticresources/blob/a3149d90ffe7ff8c0875bbb278243ae9cf02ae6a/src/plone/staticresources/profiles/default/registry/bundles.xml#L152

**Comment by davide-targa** [comment]
@thet I was not logged-in.

**Comment by davide-targa** [comment]
Check out this public form for exmple: https://bit.ly/2vOBRGb
I have no issue filling the Date and DateTime fields.
```
Plone 5.2.0 (5207)
CMF 2.4.0
Zope 4.1.1
Python 3.7.3 (default, Apr 3 2019, 05:39:12) [GCC 8.3.0]
PIL 6.1.0 (Pillow)
WSGI: On
Server: waitress 1.3.0
```

**Comment by pbauer** [comment]
@davide-targa That's weird. Can you try to do the same for 5.2.1? Maybe it it still worked in 5.2.0.

**Comment by thet** [comment]
@pbauer @davide-targa yes, 5.2.0 is probably the reason why it's working for you. We did the bundle-splitting @ ploneconf in Ferrara, where 5.2.0 was out since some months: https://github.com/plone/plone.staticresources/pull/46

... not the best semantic versioning scheme for cmfplone here ;)

**Comment by pbauer** [comment]
I just confirmed this. In Plone 5.2 the date and datetime fields still work for anonymous users. After upgrading to 5.2.1 they stop working.
I can also confirm that you can work around that by changing the condition of the bundle `plone-logged-in` in `/@@resourceregistry-controlpanel` to `python: member is not None or context.portal_type in ('EasyForm')`

**Comment by davide-targa** [comment]
@thet @pbauer I confirm that in Plone 5.2.1 Date and DateTime fields don't not work for anonymous users. Sorry for the noise.

**Comment by mactrash** [comment]
+1 same issue

**Comment by parruc** [comment]
We had this issue and solved it extending (in our own product ATM) the existing easyform bundle as follows:
- adding a js that requires mockup-patterns-pickadate
- adding a less that requires pattern.pickadate.less

If this other solution can work for collective.easyform too we would be happy to create a pull request.

**Comment by agitator** [comment]
@parruc have you tried changing the condition for the bundle to check on EasyForm as well?

If you want to help other users, you could create a separate bundle for the datepicker in staticresources and extend it's condition in the profile.xml of the EasyForm.

**Comment by parruc** [comment]
We did not need to change the condition because we included the mockup-patterns-pickadate needed by EasyForm into the easyform bundle along with other js and css we needed.

If the idea here is to move datepicker to its own bundle inside staticresources it makes absolutly sense and I will gladly share the minimal code to do it as soon as possible.

**Comment by agitator** [comment]
At least it would allow the datepicker to be enabled through the resource registry without loading the whole logged-in-bundle.
As we did that for datatables https://github.com/plone/plone.staticresources/blob/master/src/plone/staticresources/static/plone-datatables.js

@thet what do you think?

**Comment by parruc** [comment]
In the meantime I opened a ticket in plone.staticresources: https://github.com/plone/plone.staticresources/issues/96 given that the needed changes are to be done there, ok?

**Comment by vanne** [comment]
It's not only date/time fields/pickers that are not working.
All other JS that validates fields on blur is also missing.

For instance logged in users see a "Required input is missing." (when blurring a mandatory field, leaving it empty) while logged out users see nothing.

**Comment by mactrash** [comment]
Just test clean install Plone 5.2.3 the  date/time fields/pickers that are not working.

**Comment by mactrash** [comment]
Also test clean install Plone 5.2.4rc the date/time fields/pickers that are not working.

**Comment by mactrash** [comment]
5.2.4 by changing the condition of the bundle plone-logged-in in /@@resourceregistry-controlpanel to python: member is not None still not working any idea?

**Comment by fredvd** [comment]
I have created an add'on that should include the date picker pattern and also inline validation to public views. 

https://github.com/collective/collective.anonymouseditpatterns

It is a first attempt that seems to work for me locally in a plain Plone 5.2.4 coredev plone site.  I first tried to get the add'on working in a customer project, but I ran into strange issues of missing resources in the compiling bundle stage. 

The anonymouseditpatterns imports the pickadate patterns' less code, but in the mockup pattern these imports failed
until I changed bootstrap to bootstrap3:

 https://github.com/plone/mockup/blob/2ec35e1bf556b76a4b1584cd11b82ebf2c93dd40/mockup/patterns/pickadate/pattern.pickadate.less#L4-L5

when I added the add'on to a 5.2 coredev checkout, the paths worked again for /bootstrap/   :-S

I've tried to reduce the bundle size by adding js stubs, but the pickadate pattern pulls in select2.  Maybe it doesn't add more js if I also add pat-select2 to collective.anonymouseditpatterns.

**Comment by agitator** [comment]
how about adding a profile to change the condition for that bundle to `python: member is not None or context.portal_type in ('EasyForm')` within easyform?

**Comment by fredvd** [comment]
And what happens to the condition for that bundle when it is merged into the meta-bundle?

**Comment by agitator** [comment]
imho it's will be merged...

@thet ping?

we're sprinting tomorrow, as every Wednesday, join us if you need some pointers

**Comment by fredvd** [comment]
> What's the status on this issue?

@oxygenius  I think the status is what you see here on github. Do you have the same problem? Which Plone versions.  Did you try collective.anonymouseditpatterns?

**Comment by pbauer** [comment]
In Plone 6 there is still no datepicker for anonymous users. You have to enter the date manually.

**Comment by witsch** [comment]
> Hi @fredvd, works like a charm!

Same here (on 5.2.13). Thank you! 🙌
